tests/v2/validation/token/token_test.go

```diff
@@ -0,0 +1,70 @@
+package token
+
+import (
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	fv3 "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/tokens"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+const (
+	initialTokenDesc = "my-token"
+	updatedTokenDesc = "changed-token"
+)
+
+type TokenTestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	cluster      *management.Cluster
+	createdToken *fv3.Token
+	patchedToken *v3.Token
+}
+
+func (tokenTestSuite *TokenTestSuite) TearDownSuite() {
+	tokenTestSuite.session.Cleanup()
+}
+
+func (tokenTestSuite *TokenTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	tokenTestSuite.session = testSession
+
+	client, err := rancher.NewClient("", tokenTestSuite.session)
+	require.NoError(tokenTestSuite.T(), err)
+
+	tokenTestSuite.client = client
+}
+
+func (tokenTestSuite *TokenTestSuite) TestPatchToken() {
+	rancherConfig := new(rancher.Config)
+	config.LoadConfig(rancher.ConfigurationFileKey, rancherConfig)
```

Comment:
Thanks for the help and explanation @caliskanugur , really appreciate it ! - implemented this

Reply:
Andrew, I am sorry this is my bad the first version was to instantiate the initial client, since this is a test case we usually have something like the above - I've updated the snippet! 

That way we bind the sub-session to the client with the `WithSession` method.